python/hail/utils/interval.py

```diff
@@ -0,0 +1,286 @@
+        """
+        if self._end is None:
+            self._end = self.point_type._convert_to_py(self._jrep.end())
+        return self._end
+
+    @property
+    def include_start(self):
+        """True if interval is inclusive of start.
+
+        Examples
+        --------
+
+        .. doctest::
+
+            >>> interval.include_start
+            True
+
+        Returns
+        -------
+        :obj:`bool`
+        """
+        if self._include_start is None:
+            self._include_start = self._jrep.includeStart()
+        return self._include_start
+
+    @property
+    def include_end(self):
+        """True if interval is inclusive of end.
+
+        Examples
+        --------
+
+        .. doctest::
+
+            >>> interval.include_end
+            False
+
+        Returns
+        -------
+        :obj:`bool`
+        """
+        if self._include_end is None:
+            self._include_end = self._jrep.includeEnd()
+        return self._include_end
+
+    @property
+    def point_type(self):
+        """Type of each element in the interval.
+
+        Examples
+        --------
+
+        .. doctest::
+
+            >>> interval.point_type
+            tint32
+
+        Returns
+        -------
+        :class:`.Type`
+        """
+        return self._point_type
+
+    def contains(self, value):
+        """True if `value` is contained within the interval.
+
+        Examples
+        --------
+
+        .. doctest::
+
+            >>> interval.contains(5)
+            True
+
+            >>> interval.contains(6)
+            False
+
+        Parameters
+        ----------
+        value :
+            Object with type :meth:`.point_type`.
+
+        Returns
+        -------
+        :obj:`bool`
+        """
+
+        from hail.expr.expression import impute_type
+        value_type = impute_type(value)
+        if value_type != self.point_type:
+            raise TypeError("'value' is incompatible with the interval point type: %s, %s" % (value_type, self.point_type))
+        return self._jrep.contains(self.point_type._jtype.ordering(), self.point_type._convert_to_j(value))
+
+    @typecheck_method(interval=interval_type)
+    def overlaps(self, interval):
+        """True if the the supplied interval contains any value in common with this one.
+
+        Parameters
+        ----------
+        interval : :class:`.Interval`
+            Interval object with the same point type.
+
+        Returns
+        -------
+        :obj:`bool`
+        """
+
+        if self.point_type != interval.point_type:
+            raise TypeError("'interval' must have the point type %s, but found %s." % (self.point_type, interval.point_type))
+        return self._jrep.mayOverlap(self.point_type._jtype.ordering(), interval._jrep)
+
+    @classmethod
+    @typecheck_method(string=str,
+                      reference_genome=reference_genome_type)
+    def parse_locus_interval(cls, string, reference_genome='default'):
```

Comment:
we already have the expr function, we don't need this one.

Reply:
We discussed this with @cseed yesterday. I thought the decision was we needed both.

Reply:
I feel like the interface for the local objects for a given type and their corresponding expressions should be the same.  Pleasing symmetry, as @jigold would say.

Reply:
if filter_intervals takes an expression, then there's no reason to have the local thing. If you need the local thing, just `.value` it. No?

Reply:
This is more code and more docs and complexity that's duplicated and can diverge. Especially since we intend to very soon completely untangle the Python objects from Scala (no hooks into the JVM). Do we want to write a locus interval parser in Python? I don't think so...

Reply:
maybe we mark it as experimental or something? I imagine that if we need to keep this function when we move the class code to Python, then it'll have to be `hl.parse_locus_interval().value` anyway.